sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java

```diff
@@ -205,8 +205,8 @@ public QueueClientBuilder endpoint(String endpoint) {
                 this.sasTokenCredential = credential;
             }
         } catch (MalformedURLException ex) {
-            LOGGER.error("The Azure Storage Queue endpoint url is malformed. Endpoint: " + endpoint);
-            throw new IllegalArgumentException("The Azure Storage Queue endpoint url is malformed. Endpoint: " + endpoint);
+            logger.logAndThrow(new IllegalArgumentException("The Azure Storage Queue endpoint url is malformed. Endpoint: " + endpoint));
+            return null;
```

Comment:
Don't need to return null here, only use that when the code analyzer thinks there is no return case when one is needed.